### ISSUE TITLE
ty doesn't recognize `Path.joinpath`

### ISSUE THREAD

**Issue by konstin** [astral-sh/ty#646]
### Summary

ty fails to recognize that `Path().joinpath("foo")` is a `Path`, just like `Path`.

The code below fails to typecheck with `Path()`, but once I use `Path().joinpath("foo")`, the type error is not caught anymore.

```
from pathlib import Path


def run_test(uv: str) -> bool:
    return True


run_test(Path().joinpath("foo"))
```

https://play.ty.dev/3568d76c-ab51-46d5-a234-0f039b79727b

### Version

ty 0.0.1-alpha.9

**Comment by AlexWaygood** [comment]
Thanks! The `joinpath` method returns `Self` in typeshed, so I think this is just https://github.com/astral-sh/ty/issues/159

**Comment by carljm** [comment]
> I think this is just [#159](https://github.com/astral-sh/ty/issues/159)

But I think we already support explicit `Self` annotations -- what we're lacking is just implicit understanding of `Self` as the type of unannotated `self`. So I don't think it makes sense that this should be part of #159

**Comment by AlexWaygood** [comment]
It's the same as `Bar.method3()` in https://github.com/astral-sh/ty/issues/159#issuecomment-2866084949

**Comment by carljm** [comment]
Hmm I didn't realize that our support for returning `Self` was limited in that way! That's a bit surprising.
